sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -351,7 +351,7 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
   }
 
   override def close(): Unit = {
-    synchronized { loadedMaps.values.asScala.foreach(_.clear()) }
```

Comment:
Wow... this was totally wrong... If we were ever doing this we had to make sure these entries must be removed from loadedMaps as well...

Reply:
Yea - the caller was only removing the entry from loadedProviders basically - as part of the unload